tasks/libs/common/git.py

```diff
@@ -1,18 +1,36 @@
 from __future__ import annotations
 
-import os.path
+import os
+from contextlib import contextmanager
 from typing import TYPE_CHECKING
 
 from invoke.exceptions import Exit
 
-from tasks.libs.common.color import color_message
+from tasks.libs.common.color import Color, color_message
 from tasks.libs.common.constants import DEFAULT_BRANCH
 from tasks.libs.common.user_interactions import yes_no_question
 
 if TYPE_CHECKING:
     from collections.abc import Iterable
 
 
+@contextmanager
+def clone(ctx, repo, branch, destination="", options=""):
+    """
+    Context manager to clone a git repository and checkout a specific branch.
+    """
+    current_dir = os.getcwd()
+    if len(destination) == 0:
+        destination = repo
+    try:
+        ctx.run(f"git clone -b {branch} {options} https://github.com/DataDog/{repo} {destination}")
+        os.chdir(destination)
+        yield
+    finally:
+        os.chdir(current_dir)
+        ctx.run(f"rm -rf {destination}")
```

Comment:
I think you could use https://docs.python.org/3/library/tempfile.html#tempfile.TemporaryDirectory to create a temp dir instead of managing your own and taking care of deleting it

Reply:
done